### PR TITLE
[Documentation] DataConnection

### PR DESCRIPTION
**Pull request by MaceWindu**
Xml documentation for DataConnection class public surface

### CODE REVIEW

**Comment by MaceWindu** [comment]
[ci skip]

File: Source/Data/DataConnection.cs [hunk 1]
```diff
@@ -397,6 +515,12 @@ static DataConnection()
 		static readonly List<Func<IConnectionStringSettings,string,IDataProvider>> _providerDetectors =
 			new List<Func<IConnectionStringSettings,string,IDataProvider>>();
 
+		/// <summary>
+		/// Registers database provider factory method.
+		/// Factory accepts connection string settings and connection string. Could return null, of cannot create provider
```
Comment (sdanyliv) [inline Source/Data/DataConnection.cs]: Typo here 'if' not 'of'

File: Source/Data/DataConnection.cs [hunk 2]
```diff
@@ -170,17 +238,32 @@ public DataConnection(string configurationString)
...
 #endif
 
 		static readonly ConcurrentDictionary<string,int> _configurationIDs;
 		static int _maxID;
 
 		private int? _id;
+		/// <summary>
+		/// Unique identifier of mapping schema and connection combination. Shared by connections with the same mapping schema and connection configuration or connection string.
+		/// </summary>
 		public  int   ID
```
Comment (igor-tkachev) [inline Source/Data/DataConnection.cs]: This property is for internal use only. We can hide it and make it internal.

File: Source/Data/DataConnection.cs [hunk 3]
```diff
@@ -683,19 +856,29 @@ internal void InitCommand(CommandType commandType, string sql, DataParameter[] p
...
 		{
 			get { return _command ?? (_command = CreateCommand()); }
 			set { _command = value; }
 		}
 
+		/// <summary>
+		/// Creates new command object, associated with current connection/transaction.
+		/// </summary>
+		/// <returns>Command object.</returns>
 		public IDbCommand CreateCommand()
```
Comment (igor-tkachev) [inline Source/Data/DataConnection.cs]: Not sure this method should be public.

File: Source/Data/DataConnection.cs [hunk 4]
```diff
@@ -709,6 +892,9 @@ public IDbCommand CreateCommand()
 			return command;
 		}
 
+		/// <summary>
+		/// Disposes current command object, associated with connection.
+		/// </summary>
 		public void DisposeCommand()
```
Comment (igor-tkachev) [inline Source/Data/DataConnection.cs]: And this one.

File: Source/Data/DataConnection.cs (hunk 2 above)
Comment (jogibear9988) [inline Source/Data/DataConnection.cs]: if we do so, we should increase major version, it's maybe a breaking change

**Comment by MaceWindu** [comment]
- fixed typo
- marked internal API methods with "Internal api" comment for now
- examples will be provided with next PR

**Pull request merged**